parsers/KPX.py

```diff
@@ -165,27 +165,15 @@ def parse_chart_prod_data(
 
         dt = TIMEZONE.localize(datetime.strptime(item["regDate"], "%Y-%m-%d %H:%M"))
 
-        production_mix = ProductionMix()
-        production_mix.add_value(
-            "coal",
-            round(float(item["coal"]) + float(item["localCoal"]), 5),
+        production_mix = ProductionMix(
+            coal=float(item["coal"]) + float(item["localCoal"]),
+            gas=float(item["gas"]),
+            hydro=float(item["waterPower"]),
+            nuclear=float(item["nuclearPower"]),
+            oil=float(item["oil"]),
+            unknown=float(item["newRenewable"]),
```

Comment:
Sorry this is a bit late but we should avoid assigning values like this directly as they don't get the same validation checks then. (Right now it will not correct for negative values if they are present or correct floating point errors.)

Reply:
Ah shit only saw your comment now - do you have time to make a quick PR to fix that? I'm planning on releasing a new version in the morning so we could have that in there

Reply:
> Ah shit only saw your comment now - do you have time to make a quick PR to fix that? I'm planning on releasing a new version in the morning so we could have that in there

I actually just double checked and the negative values will be corrected. But some of the other validation we do will not. So feel free to merge this and I'll open a new PR during the day.